src/test/java/org/commcare/formplayer/services/MediaMetaDataServiceTest.java

```diff
@@ -1,83 +1,145 @@
-                "contentType",
+                "jpg",
                 4,
                 "username",
                 "asUser",
                 "domain",
                 "appid"
         );
+        when(mediaMetaDataRepo.save(any())).thenAnswer(new Answer<MediaMetadataRecord>() {
+            @Override
+            public MediaMetadataRecord answer(InvocationOnMock invocation) throws Throwable {
+                MediaMetadataRecord mediaMetaData = (MediaMetadataRecord)invocation.getArguments()[0];
+                if (mediaMetaData.getId() == null) {
+                    // this is normally taken care of by Hibernate
+                    ReflectionTestUtils.setField(mediaMetaData, "id", UUID.randomUUID().toString());
+                }
+                return mediaMetaData;
+            }
+        });
+
+        when(mediaMetaDataRepo.findById(any())).thenAnswer(new Answer<Optional<MediaMetadataRecord>>() {
+            @Override
+            public Optional<MediaMetadataRecord> answer(InvocationOnMock invocation) throws Throwable {
+                return Optional.ofNullable(mediaMetaData);
+            }
+        });
+
+        doAnswer(invocation -> {
+            mediaMetaData = null;
+            return null;
+        }).when(mediaMetaDataRepo).deleteById(any());
+
+        when(mediaMetaDataRepo.getFormSessionIsNull()).thenAnswer(new Answer<List<MediaMetadataRecord>>() {
+            @Override
+            public List<MediaMetadataRecord> answer(InvocationOnMock invocation) throws Throwable {
+                List<MediaMetadataRecord> recordsList = new ArrayList<>();
+                recordsList.add(mediaMetaData);
+                return recordsList;
+            }
+        });
+    }
+
+    @AfterEach
+    public void tearDown() {
+        testFile.delete();
     }
 
     @Test
     public void testSaveMetaData() {
         mediaMetaDataService.saveMediaMetaData(mediaMetaData);
-
         Optional<MediaMetadataRecord> fetchedMediaMetaData = mediaMetaDataRepo.findById(mediaMetaData.getId());
-
         Assertions.assertTrue(fetchedMediaMetaData.isPresent());
     }
 
     @Test
     public void testDelete() {
         mediaMetaDataService.saveMediaMetaData(mediaMetaData);
-
         Optional<MediaMetadataRecord> fetchedMediaMetaData = mediaMetaDataRepo.findById(mediaMetaData.getId());
-
         Assertions.assertTrue(fetchedMediaMetaData.isPresent());
+        String metadataId = mediaMetaData.getId();
+        mediaMetaDataService.deleteMetaDataById(metadataId);
+        Optional<MediaMetadataRecord> newlyFetchedMediaMetaData = mediaMetaDataRepo.findById(metadataId);
+        Assertions.assertFalse(newlyFetchedMediaMetaData.isPresent());
+    }
 
-        mediaMetaDataService.deleteMetaDataById(mediaMetaData.getId());
+    @Test
+    public void testPurge() {
+        Assertions.assertTrue(testFile.exists());
+        Integer purgeCount = mediaMetaDataService.purge(Instant.now());
+        Assertions.assertEquals(1, purgeCount);
```

Comment:
also verify `testFile.exisits()` is false now

Reply:
I'll go ahead and add that assert. purgeCount is not passing for some reason. still working on why.